packages/realm_dart/test/realm_logger_test.dart

```diff
@@ -1,211 +1,115 @@
 // Copyright 2023 MongoDB, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-// ignore_for_file: unused_local_variable, avoid_relative_lib_imports
-
 import 'dart:async';
 import 'dart:isolate';
 import 'package:logging/logging.dart';
+import 'package:realm_dart/src/logging.dart';
 import 'package:test/test.dart' hide test, throws;
-import 'package:realm_dart/src/realm_class.dart' show RealmInternal;
 import 'package:realm_dart/realm.dart';
 import 'test.dart';
 
-const logLevels = [
-  RealmLogLevel.all,
+const logToValues = [
   RealmLogLevel.trace,
   RealmLogLevel.debug,
   RealmLogLevel.detail,
   RealmLogLevel.info,
-  RealmLogLevel.fatal,
+  RealmLogLevel.warn,
   RealmLogLevel.error,
-  RealmLogLevel.warn
+  RealmLogLevel.fatal,
 ];
 
-class LoggedMessage {
-  final Level level;
-  final String message;
-
-  const LoggedMessage(this.level, this.message);
-
-  factory LoggedMessage.empty() => const LoggedMessage(RealmLogLevel.off, "");
-
-  @override
-  // ignore: hash_and_equals
-  bool operator ==(Object other) {
-    if (identical(this, other)) return true;
-    if (other is! LoggedMessage) return false;
-    return level == other.level && message == other.message;
-  }
-
-  @override
-  String toString() => "level:$level message:$message";
-}
-
-void openARealm() {
-  Configuration.defaultRealmName = generateRandomRealmPath();
-  final config = Configuration.inMemory([Car.schema]);
-  Realm(config).close();
-  tryDeleteRealm(Configuration.defaultRealmName);
-}
-
 void main() {
   setupTests();
 
-  for (var level in logLevels) {
-    test('Realm.logger supports log level $level', () async {
-      LoggedMessage actual = await Isolate.run(() async {
-        final completer = Completer<LoggedMessage>();
-
-        Realm.logger.level = level;
-        await Future<void>.delayed(const Duration(milliseconds: 10));
-        Realm.logger.onRecord.listen((event) {
-          completer.complete(LoggedMessage(event.level, event.message));
-        });
-
-        RealmInternal.logMessageForTesting(level, "123");
-
-        return await completer.future;
+  group('All levels', () {
+    for (var level in RealmLogLevel.values) {
+      test('Realm.logger supports log level $level', () {
+        Realm.logger.setLogLevel(RealmLogLevel.off);
+        Realm.logger.setLogLevel(RealmLogLevel.all, category: RealmLogCategory.realm.sdk);
```

Comment:
Why is the `off` necessary here?

Reply:
It suppress all other trace than what is send to `RealmLogCategory.realm.sdk` ..like sync stuff in background.